src/uploader/hugging_face_uploader.py

```
@@ -0,0 +1,145 @@
+import io
+import huggingface_hub
+from huggingface_hub.utils import RepositoryNotFoundError, RevisionNotFoundError
+from fastapi import HTTPException, UploadFile, status
+from requests import HTTPError
+from sqlalchemy.engine import Engine
+from sqlalchemy.orm import joinedload
+from sqlmodel import Session
+
+from database.model.dataset.data_download import DataDownloadORM
+from database.model.dataset.dataset import Dataset
+
+
+class HuggingfaceUploader:
+    def __init__(self, engine: Engine):
+        self.engine = engine
+
+    def handle_upload(
+        self,
+        identifier: int,
+        file: UploadFile,
+        token: str,
+        username: str,
+    ):
+        dataset = self._get_resource(engine=self.engine, identifier=identifier)
+        repo_id = f"{username}/{dataset.name}"
+
+        url = self._create_or_get_repo_url(repo_id, token)
+
+        try:
+            huggingface_hub.upload_file(
+                path_or_fileobj=io.BufferedReader(file.file),
+                path_in_repo=f"/data/{file.filename}",
+                repo_id=repo_id,
+                repo_type="dataset",
+                token=token,
+            )
+        except HTTPError as e:
+            msg = f"Error uploading the file, huggingface api returned a http error: {e.strerror}"
+            raise HTTPException(status_code=status.HTTP_502_BAD_GATEWAY, detail=msg)
+
+        except ValueError:
+            msg = "Error uploading the file, bad format"
```

You're now completely ignoring the error message. I think that it would be nice to give some error message back to the user in all error messages, similar as you do for (some) of the other errors.

There might be information in the HuggingFace error message (field "foo" is required) and that's completely ignored now.